### DISCUSSION TITLE
multi-select field getting truncated to 1 value when writing to MySQL

### DISCUSSION THREAD

**Discussion by BillThyCat** [Budibase/budibase#6507] [Help]
I gave up on the previous design of my time sheet app, I now have all of my tables stored in MySQL and MS-SQL and I'm able to query against it to find the employee's I want to list based on the foreman assigned to them.  This is all working now, however....

On my previous version of the app when writing the row budibase recognized the multi-select type and accepted multiple selections into the database.  MySQL not so much, it only stores the first value from the multi-select tool.  I created a card to prove that the field contained more than 1 selection, so I know it's correct, it's just when it writes it it only sends one.

I had it set to text originally and then saw someone say it had to be long text so I changed it to that, and still only getting the first value.

Is there a javascript  that I could write that would just make it a long string right before it writes it to the row?  Maybe using the update field value option under define actions?

I'm going to have a python script that reads the data later to create the import file, so I don't care how it gets stored, just as long as all of the employee ID's are there so it can parse them.

Thanks, I do really appreciate the responses that I get.

**Comment by melohagan** [comment]
Hi @BillThyCat 

I think I had a look at this issue, or something similar, a while back: https://github.com/Budibase/budibase/issues/5204

At the time I came to the conclusion that you needed to use the Relationship Picker: https://docs.budibase.com/docs/pickers#relationship-picker

**Reply by BillThyCat** [reply]
I had nothing but trouble trying to use the relationship picker, so I opted to just create a 3rd database that contains 2 fields, foreman_id, and a single employee_id, then I run a query against it and the results are what I put into the multi-picker and then I can select multiple people and as I said, if I put the results into a card they display all of the selected users, but when writing the row it just writes the first user.

I only am picking 1-3 users, so it's not like it's to large of a field size to be written to it.

![image](https://user-images.githubusercontent.com/83614062/176924774-6172e783-5db6-4e45-b800-b7012bdea8e8.png)

These are all the fields and their types in the database.

Is there any javascripts you know of that can take all of the selections from a multi-select tool and just combine them into a single string?  

Thanks,

**Reply by melohagan** [reply]
Well you can simply use `.toString()` on an array in JavaScript.

**Reply by BillThyCat** [reply]
Is the field value of the multi-select treated as an array?  And does the one command dump the whole array, or do I have to address each value [0],[1],[2] etc as part of the script?
or do I just do something like newstring=multiselect.field.tostring()  and I would assume I need to create a new field that matches the name of one of the fields in the database and as I press the write row button I call the script first to populate that field?

**Comment by melohagan** [comment]
You may want to try something like this:

![Screenshot 2022-07-01 at 17 30 53](https://user-images.githubusercontent.com/101575380/176934539-46d97fe8-fa7c-4204-b94a-6e71e162d891.png)

![Screenshot 2022-07-01 at 17 37 33](https://user-images.githubusercontent.com/101575380/176935463-f2f4684e-75e3-40dd-ae3c-0f940c0cbb69.png)

**JavaScript**
`return $("Form.Fields.Multi")?.toString();`

**Reply by BillThyCat** [reply]
Not working.  I tried doing it on change of the multi-select tool, I created a text field and set it's default value to the javascript, I created a card and had it display it as javascript and none of them seemed to accept that script.  No errors, but just didn't seem to work.
Writing the row still only wrote the single value.

What does the ? between .Multi") and the .toString do?  

Any other scripts that would do something similar?

**Reply by BillThyCat** [reply]
Ok, if I put the script in a card and use the .join(','); command it works, but only inside of a card.  Not sure what variable type is used in a card, but it's not working in a text box.

Any way to read a value from inside of a card and put that into a field before writing to the database?

Handlebars don't seem to be able to directly address title, subtitle, etc.

Starting to make progress.....

Thanks,

**Reply by BillThyCat** [reply]
Ok, using a javascript

```
const empList=$("New Form.Fields.employees2")?.join(',');
return empList.toString();

```

It seems to work, tho it lags on getting the last value, so if I choose 3 it only displays 2, however if I make every other field just update the field assuming they will be picking at least 1 other field, then it seems to update.  However if they have the whole form filled in and then decide to go up and add or remove one more person, it's not always getting updated.  I even have it on the button action that writes the row to run the script again, validate the whole form, then write the row, hoping it triggers off correctly and updates it, but unfortunately it's not flawless.  And the person entering time isn't going to notice that it didn't have them there correctly.

I'm at least making good progress, may need to have it pop up a modal window to confirm that all of the users are listed and require an input from the user to then write the row.

Thanks,

**Reply by melohagan** [reply]
Hey @BillThyCat The lag you speak of, as a hunch it could be that you are not using the correct value binding for the On Change.
https://docs.budibase.com/docs/actions#on-change

If you check out that link, you will see that the binding is `{{ Field Value }}`. This represents the value(s) that was selected on change. 
The JavaScript binding should be selectable on the left hand pane, and should be `$("Field Value")`
If you use `$("New Form.Fields.employees2")` then it will get the previous value (before the on change was triggered) hence the lag.

Also the `?` in JavaScript is a null propagation operator. Basically it just means if the value happens to be null/undefined then don't try to call the method because it would error otherwise.
